x/tss/keeper/msg_server.go

```diff
@@ -135,20 +135,19 @@ func (s msgServer) HeartBeat(c context.Context, req *types.HeartBeatRequest) (*t
 		s.Logger(ctx).Error(fmt.Sprintf("validator %s not ready to participate in signing due to: %s", valAddr.String(), response.SigningIllegibility.String()))
 	}
 
-	// metrics for heartbeat
-	labels := []metrics.Label{
-		telemetry.NewLabel("timestamp", strconv.FormatInt(ctx.BlockTime().Unix(), 10)),
-		telemetry.NewLabel("address", valAddr.String()),
-		telemetry.NewLabel("height", strconv.FormatInt(ctx.BlockHeight(), 10)),
-		telemetry.NewLabel("key_IDs", strings.Join(exported.KeyIDsToStrings(req.KeyIDs), ",")),
-	}
-
-	//illegibilities
+	// metrics
 	for _, ill := range snapshot.GetValidatorIllegibilities() {
-		labels = append(labels, telemetry.NewLabel(strings.ReplaceAll(ill.String(), "-", "_"), strconv.FormatBool(illegibility.Is(ill))))
+		gauge := float32(0)
+		if illegibility.Is(ill) {
+			gauge = 1
+		}
+
+		telemetry.SetGaugeWithLabels([]string{types.ModuleName, "illegibility_" + ill.String()},
+			gauge, []metrics.Label{telemetry.NewLabel("address", valAddr.String())})
 	}
 
-	telemetry.SetGaugeWithLabels([]string{types.ModuleName, "heartbeat"}, 1, labels)
+	telemetry.SetGaugeWithLabels([]string{types.ModuleName, "heartbeat_last_block_sent"},
+		float32(ctx.BlockHeight()), []metrics.Label{telemetry.NewLabel("address", valAddr.String())})
```

Comment:
don't we have any other way to do this (maybe outside the telemetry package)? the float32 cast will make the metric useless in no time

Reply:
prometheus should support float64